### PR TITLE
feat(cloudquotas): generate library

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394)

### CODE REVIEW

File: generator/generator_config.textproto [hunk 1]
```diff
@@ -582,6 +582,14 @@ service {
   emitted_rpcs: ["RunReportJob", "FetchReportResults", "ListReports"]
 }
 
+# Cloud Quotas
+service {
+  service_proto_path: "google/api/cloudquotas/v1/cloudquotas.proto"
+  product_path: "google/cloud/cloudquotas/v1"
+  initial_copyright_year: "2023"
```
Comment (coryan) [inline generator/generator_config.textproto:589]: s/2023/2024/ and then regenerate.  I think, maybe I am wrong, check the policy if you disagree.

File: google/cloud/cloudquotas/quickstart/quickstart.cc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! [all]
+#include "google/cloud/cloudquotas/v1/cloud_quotas_client.h"
+#include "google/cloud/location.h"
```
Comment (coryan) [inline google/cloud/cloudquotas/quickstart/quickstart.cc:17]: This is unused, but maybe you could use it.

File: google/cloud/cloudquotas/v1/.repo-metadata.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "api_id": "cloudquotas.googleapis.com",
+    "api_shortname": "cloudquotas",
+    "client_documentation": "https://cloud.google.com/cpp/docs/reference/cloudquotas/latest",
+    "distribution_name": "google-cloud-cpp",
+    "issue_tracker": "https://issuetracker.google.com/issues?q=componentid:445904%20status=open",
+    "language": "cpp",
+    "library_type": "GAPIC_AUTO",
+    "name_pretty": "Cloud Quotas API",
+    "product_documentation": "https://cloud.google.com/docs/quota/api-overview",
```
Comment (coryan) [inline google/cloud/cloudquotas/v1/.repo-metadata.json:10]: This still says "preview", but I suspect that is a bug in the documentation site.

Comment (coryan) [reply google/cloud/cloudquotas/v1/.repo-metadata.json:10]: No action required.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394)* status: 0 of 55 files reviewed, 4 unresolved discussions (waiting on @coryan)

___
*[`generator/generator_config.textproto` line 589 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394#gh-1441099496:-NnGjyXm3mNtHRZa_NFA:b-bu91xm) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e73ba3f51190b1ea5fb4146becd07686d7fb8424/generator/generator_config.textproto#L589)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

s/2023/2024/ and then regenerate.  I think, maybe I am wrong, check the policy if you disagree.
</blockquote></details>

Updated
___
*[`google/cloud/cloudquotas/quickstart/quickstart.cc` line 17 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394#gh-1441100216:-NnGk-wMAougF-k8X-on:b-m9u39i) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e73ba3f51190b1ea5fb4146becd07686d7fb8424/google/cloud/cloudquotas/quickstart/quickstart.cc#L17)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

This is unused, but maybe you could use it.
</blockquote></details>

Now with more use!
___
*[`google/cloud/cloudquotas/quickstart/quickstart.cc` line 31 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394#gh-1441100828:-NnGk34H0fTucslZZ-bK:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e73ba3f51190b1ea5fb4146becd07686d7fb8424/google/cloud/cloudquotas/quickstart/quickstart.cc#L31)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Consider:

```suggestion
  auto const parent = google::cloud::Location(argv[1], "global").FullName();
```

</blockquote></details>

Done.
___
*[`google/cloud/cloudquotas/v1/.repo-metadata.json` line 10 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394#gh-1441101427:-NnGk4fl2Q7eO0vpDF10:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e73ba3f51190b1ea5fb4146becd07686d7fb8424/google/cloud/cloudquotas/v1/.repo-metadata.json#L10)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

No action required.
</blockquote></details>

Done.

File: google/cloud/cloudquotas/quickstart/quickstart.cc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    std::cerr << "Usage: " << argv[0] << " project-id\n";
+    return 1;
+  }
+
+  namespace cloudquotas = ::google::cloud::cloudquotas_v1;
+  auto client =
+      cloudquotas::CloudQuotasClient(cloudquotas::MakeCloudQuotasConnection());
+
+  auto const parent = google::cloud::Location(argv[1], "global").FullName();
+  ;
```
Comment (devbww) [inline google/cloud/cloudquotas/quickstart/quickstart.cc:32]: Empty statement?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394)* status: 0 of 55 files reviewed, 5 unresolved discussions (waiting on @coryan and @devbww)

___
*[`google/cloud/cloudquotas/quickstart/quickstart.cc` line 32 at r4](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394#gh-1441122547:-NnGlZnbDs9wNLTSyoEL:bss4a2b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/6928d8e26d92a28185b71bbba233482055d41f58/google/cloud/cloudquotas/quickstart/quickstart.cc#L32)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Empty statement?
</blockquote></details>

Removed.

**Review by coryan — APPROVED** [review]
> Reviewed 55 of 55 files at r1, 35 of 36 files at r3, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13394)* status: all files reviewed, 1 unresolved discussion (waiting on @devbww)

**Pull request merged**